sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/viewmodel/ScheduleRepository.kt

```diff
@@ -0,0 +1,218 @@
+import org.sagebionetworks.research.sageresearch.dao.room.ScheduledActivityEntityDao
+import org.sagebionetworks.research.sageresearch.util.SingletonWithParam
+
+import rx.Observable
+import rx.schedulers.Schedulers
+
+import java.util.TreeMap
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+
+//
+//  Copyright © 2018 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+/**
+ * The ScheduleRepository is responsible for downloading the study's schedules and saving them to the Room database
+ * All @see ScheduleViewModel are required to call this class' function syncSchedules() to ensure that
+ * the schedules are currently synced with bridge.
+ */
+open class ScheduleRepository(context: Context) {
+
+    /**
+     * Creates a singleton that takes Context as a parameter to access
+     * Use ScheduleRepository.getInstance(context)
+     * TODO: mdephillips 9/2/18 use dagger?
+     */
+    companion object : SingletonWithParam<ScheduleRepository, Context>({
+        ScheduleRepository(it.applicationContext)
+    })
+
+    private val entityConverter = EntityTypeConverters()
+
+    private val scheduleDao: ScheduledActivityEntityDao =
+            ResearchDatabase.getInstance(context).scheduleDao()
+
+    private var isSyncing = AtomicBoolean(false)
```

Comment:
Would it be nice to expose this as a LiveData? Thinking UI may display sync states

Reply:
Ohhh that's a great idea!